src/mappings/Competition/mapping.ts

```diff
@@ -100,6 +140,31 @@ export function handleNewVote(event: NewVote): void {
       suggestion.save();
     }
     vote.save();
+
+    let competition = Competition.bind(event.address);
+    competitionProposal.winningSuggestions = [];
+    let winningSuggestions = competitionProposal.winningSuggestions;
+    debug(competitionProposal.suggestions.toString() + '');
+    for (let i = 0; i < competitionProposal.suggestions.length; i++) {
+      let suggestions = competitionProposal.suggestions;
+      if (suggestions != null) {
+        let currentSuggestionId = (suggestions as string[])[i];
+        let competitionSuggestion = CompetitionSuggestion.load(currentSuggestionId as string);
+        if (competitionSuggestion != null) {
+          let index = competition.getOrderedIndexOfSuggestion(competitionSuggestion.suggestionId);
```

Comment:
it is not optimised to call the contract in a loop for all suggestions.
all this can be avoided by sorting the suggestions.